### PR TITLE
CIS: GLB health check request headers support issue#1844

### PR DESCRIPTION
**Pull request by KumarGanesanIBM**

### CODE REVIEW

File: ibm/data_source_ibm_cis_healthchecks.go [hunk 1]
```diff
@@ -115,6 +115,26 @@ func dataSourceIBMCISHealthChecks() *schema.Resource {
 							Type:     schema.TypeInt,
 							Computed: true,
 						},
+						cisGLBHealthCheckHeader: {
+							Type:     schema.TypeSet,
```
Comment (kavya498) [inline ibm/data_source_ibm_cis_healthchecks.go:119]: It is advisable to have typeList for datasources..
Please chage it to typeSet in datasource as it is easily accessable..

File: ibm/resource_ibm_cis_healthcheck.go [hunk 2]
```diff
@@ -24,6 +24,9 @@ const (
 	cisGLBHealthCheckAllowInsecure   = "allow_insecure"
 	cisGLBHealthCheckCreatedOn       = "create_on"
 	cisGLBHealthCheckModifiedOn      = "modified_on"
+	cisGLBHealthCheckHeader          = "header"
```
Comment (kavya498) [inline ibm/resource_ibm_cis_healthcheck.go:27]: `headers` would be more readable that `header`

File: ibm/data_source_ibm_cis_healthchecks.go [hunk 3]
```diff
@@ -115,6 +115,26 @@ func dataSourceIBMCISHealthChecks() *schema.Resource {
...
+									cisGLBHealthCheckHeaderValues: {
+										Type:     schema.TypeSet,
+										Computed: true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+								},
+							},
+							Set: hashByMapKey(cisGLBHealthCheckHeaderHeader),
```
Comment (kavya498) [inline ibm/data_source_ibm_cis_healthchecks.go:136]: not required if type list is used

File: website/docs/r/cis_healthcheck.html.markdown [hunk 4]
```diff
@@ -43,6 +43,7 @@ The following arguments are supported:
 - `allow_insecure`-(Optional,bool) Do not validate the certificate when healthcheck use HTTPS.
 - `description` - (Optional,string) Free text description.The Default value is false.
 - `port` - (Optional,int) The TCP port to use for the health check.
+- `header` - (Optional,string) The health check header.
```
Comment (kavya498) [inline website/docs/r/cis_healthcheck.html.markdown:46]: add elements of header `header` and `values` in docs

File: ibm/data_source_ibm_cis_healthchecks.go (hunk 1 above)
Comment (KumarGanesanIBM) [reply ibm/data_source_ibm_cis_healthchecks.go:119]: it did not work after changing it to TypeList

File: ibm/resource_ibm_cis_healthcheck.go (hunk 2 above)
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_cis_healthcheck.go]: done

File: ibm/data_source_ibm_cis_healthchecks.go (hunk 3 above)
Comment (KumarGanesanIBM) [reply ibm/data_source_ibm_cis_healthchecks.go]: it did not work after changing to TypeList

File: website/docs/r/cis_healthcheck.html.markdown (hunk 4 above)
Comment (KumarGanesanIBM) [reply website/docs/r/cis_healthcheck.html.markdown]: done

File: ibm/data_source_ibm_cis_healthchecks.go (hunk 1 above)
Comment (kavya498) [reply ibm/data_source_ibm_cis_healthchecks.go:119]: Hi..
When changed to type list `Set: hashByMapKey(cisGLBHealthCheckHeadersHeader),` has to be removed but schema
seperate flatten function should be written for datasource that returns list

Comment (KumarGanesanIBM) [reply ibm/data_source_ibm_cis_healthchecks.go]: modified

File: ibm/data_source_ibm_cis_healthchecks.go (hunk 3 above)
Comment (KumarGanesanIBM) [reply ibm/data_source_ibm_cis_healthchecks.go]: modified

**Pull request merged**